qemu/tests/live_snapshot_basic.py

```diff
@@ -0,0 +1,116 @@
+
+        :param params: A dict containing VM preprocessing parameters.
+        :param env: The environment (a dict-like object).
+        """
+        self.params = params
+        self.env = env
+        self.vm = env.get_vm(params["main_vm"])
+        self.snapshot_file = params.get("snapshot_file")
+        self.snapshot_format = params.get("snapshot_format", "qcow2")
+        self.snapshot_mode = params.get("snapshot_mode", "absolute-paths")
+        self.node_name = params.get("node_name")
+        self.snapshot_node_name = params.get("snapshot_node_name")
+
+    def create_image(self):
+        """
+        Create a image.
+        """
+        image_name = self.params.get("image_name")
+        self.params['image_name_snapshot'] = image_name + "-snap"
+        image_name_snapshot = self.params['image_name_snapshot']
+        snapshot_params = self.params.object_params("snapshot")
+        base_dir = self.params.get("images_base_dir", data_dir.get_data_dir())
+
+        image_io = QemuImg(snapshot_params, base_dir, image_name)
+        image_name, _ = image_io.create(snapshot_params)
+        return image_name
+
+    def get_base_image(self):
+        """
+        Get base image.
+        """
+        base_file = storage.get_image_filename(self.params,
+                                               data_dir.get_data_dir())
+        return self.vm.get_block({"file": base_file})
+
+    def get_snapshot_file(self):
+        """
+        Get path of snapshot file.
+        """
+        snapshot_file = "images/%s" % self.snapshot_file
+        return utils_misc.get_path(data_dir.get_data_dir(), snapshot_file)
+
+    def create_snapshot(self):
+        """
+        Create a live disk snapshot.
+        """
+        if self.snapshot_mode == "existing":
+            error_context.context("Creating an image ...", logging.info)
+            self.snapshot_file = self.create_image()
+        else:
+            self.snapshot_file = self.get_snapshot_file()
+        device = self.get_base_image()
+        snapshot_args = {"format": self.snapshot_format,
+                         "mode": self.snapshot_mode}
+        if self.node_name:
+            snapshot_args.update({"node_name": self.node_name})
+        if self.snapshot_node_name:
+            snapshot_args.update({"snapshot_node_name": self.snapshot_node_name})
+        self.vm.monitor.live_snapshot(device, self.snapshot_file,
+                                      **snapshot_args)
+
+    def check_snapshot(self):
+        """
+        Check whether the snapshot is created successfully.
+        """
+        snapshot_info = str(self.vm.monitor.info("block"))
+        if self.snapshot_file not in snapshot_info:
+            logging.error(snapshot_info)
+            raise exceptions.TestFail("Snapshot doesn't exist")
+        if self.snapshot_node_name and (self.snapshot_node_name not in
+                                        snapshot_info):
```

Comment:
Here should check the value of 'node_name' field from 'query-block' output equaling the value set in cfg file.